### PR TITLE
[LLM Runtime] Add python API for GGUF

### PR DESCRIPTION
**Pull request by Zhenzhong1**
## Type of Change

API extend & Doc update

## Description

To directly load a GGUF model from HF, here is the sample code:
```python
from transformers import AutoTokenizer, TextStreamer
from intel_extension_for_transformers.transformers import AutoModelForCausalLM, WeightOnlyQuantConfig

# Specify the GGUF repo on the Hugginface
model_name = "TheBloke/Llama-2-7B-Chat-GGUF"
# Download the the specific gguf model file from the above repo
model_file = "llama-2-7b-chat.Q4_0.gguf"
# make sure you are granted to access this model on the Huggingface.
tokenizer_name = "meta-llama/Llama-2-7b-chat-hf"

# If you are not granted, please check this combination.
# model_name = "TheBloke/Mistral-7B-v0.1-GGUF"
# model_file = "mistral-7b-v0.1.Q4_0.gguf" 
# tokenizer_name = "mistralai/Mistral-7B-v0.1"

prompt = "Once upon a time"
tokenizer = AutoTokenizer.from_pretrained(tokenizer_name, trust_remote_code=True)
inputs = tokenizer(prompt, return_tensors="pt").input_ids
streamer = TextStreamer(tokenizer)
model = AutoModelForCausalLM.from_pretrained(model_name, model_file = model_file)
outputs = model.generate(inputs, streamer=streamer, max_new_tokens=300)
```
TheBloke/Llama-2-7B-Chat-GGUF screenshot:
image

TheBloke/Mistral-7B-v0.1-GGUF screenshot:
image

Validated GGUF models from HuggnigFace: [llama2-7b](https://huggingface.co/TheBloke/Llama-2-7B-Chat-GGUF), [Mistral-7B](https://huggingface.co/TheBloke/Mistral-7B-v0.1-GGUF), [CodeLlama-7B](https://huggingface.co/TheBloke/CodeLlama-7B-GGUF), [CodeLlama-13B](https://huggingface.co/TheBloke/CodeLlama-13B-GGUF).

If you want to load GGUF files created by personal or llama.cpp, please check the https://github.com/intel/neural-speed/pull/48 for more details.

## Expected Behavior & Potential Risk

N/A

## How has this PR been tested?

Manual

## Dependency Change?

from huggingface_hub import hf_hub_download

### CODE REVIEW

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 1]
```diff
@@ -276,6 +276,26 @@ model = AutoModelForCausalLM.from_pretrained(model_name, quantization_config=woq
...
+# Download the the specific gguf model file from the above repo
+model_file = "llama-2-7b-chat.Q4_0.gguf"
+# make sure you are granted to access this model on the Huggingface.
+tokenizer_name = "meta-llama/Llama-2-7b-chat-hf"
+
+prompt = "Once upon a time"
+tokenizer = AutoTokenizer.from_pretrained(tokenizer_name, trust_remote_code=True)
+inputs = tokenizer(prompt, return_tensors="pt").input_ids
+streamer = TextStreamer(tokenizer)
+model = AutoModelForCausalLM.from_pretrained(model_name = model_name, model_file = model_file)
```
Comment (a32543254) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:295]: Nice! Certainly what we want API

File: intel_extension_for_transformers/transformers/modeling/modeling_auto.py [hunk 2]
```diff
@@ -117,7 +117,31 @@ class _BaseQBitsAutoModelClass:
     ORIG_MODEL = None
 
     @classmethod
-    def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
+    def from_pretrained(cls, pretrained_model_name_or_path = None, *model_args, **kwargs):
```
Comment (a32543254) [inline intel_extension_for_transformers/transformers/modeling/modeling_auto.py:120]: why set pretrained_model_name_or_path default as none?

Comment (Zhenzhong1) [reply intel_extension_for_transformers/transformers/modeling/modeling_auto.py]: fixed

**Review by intellinjun — APPROVED** [review]

**Comment by VincyZhang** [comment]
Neuralchat UT is irrelevant.

**Comment by VincyZhang** [comment]
@Zhenzhong1  Could you please add UT here later? https://github.com/intel/intel-extension-for-transformers/tree/main/intel_extension_for_transformers/llm/runtime/graph/tests

**Pull request merged**